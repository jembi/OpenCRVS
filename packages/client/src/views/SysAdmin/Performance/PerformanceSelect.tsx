/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import * as React from 'react'
import {
  ISelect2Option,
  ISelect2Props,
  Select2
} from '@opencrvs/components/lib/Select/Select2'

export interface IPerformanceSelectOption extends ISelect2Option {
  type?: string
}
interface IOperationalSelectProps
  extends ISelect2Props<IPerformanceSelectOption> {}

export function PerformanceSelect(props: IOperationalSelectProps) {
  return <Select2 {...props} />
}
