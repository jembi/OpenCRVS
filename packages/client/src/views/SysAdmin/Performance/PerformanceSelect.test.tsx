/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import * as React from 'react'
import { createShallowRenderedComponent } from '@client/tests/util'
import { PerformanceSelect } from './PerformanceSelect'

describe('PerformanceSelect tests', () => {
  it('shallow rendered component matches snapshot', () => {
    const shallowPerformanceSelect = createShallowRenderedComponent(
      <PerformanceSelect
        value="OPTION1"
        options={[
          { label: 'Option 1', value: 'OPTION1' },
          { label: 'Option 2', value: 'OPTION2', type: 'TYPE' }
        ]}
      />
    )

    expect(shallowPerformanceSelect).toMatchSnapshot()
  })
})
