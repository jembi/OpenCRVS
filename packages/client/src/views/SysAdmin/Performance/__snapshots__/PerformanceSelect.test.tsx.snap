// Vitest Snapshot v1

exports[`PerformanceSelect tests > shallow rendered component matches snapshot 1`] = `ShallowWrapper {}`;

exports[`PerformanceSelect tests shallow rendered component matches snapshot 1`] = `ShallowWrapper {}`;
